rebase_automation.sh

```
@@ -0,0 +1,527 @@
+    
+    if [[ "$DRY_RUN" == "true" ]]; then
+        log_warning "[DRY RUN] Would update Makefile:"
+        echo "  BUNDLE_VERSION: $OLD_BUNDLE_VERSION -> $NEW_BUNDLE_VERSION"
+        echo "  CERT_MANAGER_VERSION: v$OLD_CERT_MANAGER_VERSION -> v$NEW_CERT_MANAGER_VERSION"
+        echo "  CHANNELS: stable-v1,stable-v$(echo $OLD_BUNDLE_VERSION | cut -d'.' -f1,2) -> stable-v1,stable-v$(echo $NEW_BUNDLE_VERSION | cut -d'.' -f1,2)"
+        echo "  Would run: make update && make bundle"
+        return 0
+    fi
+    
+    # Extract major.minor versions for channels
+    local old_channel_version=$(echo "$OLD_BUNDLE_VERSION" | cut -d'.' -f1,2)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)